### DISCUSSION TITLE
Access Initialized IState<TState> in OnInitialized in Page/Component

### DISCUSSION THREAD

**Discussion by gcalabro-rli** [mrpmorris/Fluxor#508] [Q&A]
If I try to do DI `TState` and `OnInitialized` try to access current state, it's not yet initialized and it's in a default state. Is there something I am missing or a better way to do? I know I could `+=` to `StateHasChanged` and do things but I was hoping to avoid that here.

```
[Inject]
public required IState<TState> MyState { get; set; }

protected override void OnInitialized()
{
    base.OnInitialized();
    Console.WriteLine($">>>{MyState.Value.Foo}");
}
```

**Comment by mrpmorris** [comment]
Have you remembered to use StoreInitializer?
